backend/flask_main.py

```diff
@@ -66,7 +66,7 @@ def login():
 
     try:
         user = get_user_by_email(user_email, redis_client)
-    except TypeError:  # TODO implement custom exception type
+    except EntryNotFoundError:
```

Comment:
this function can throw either this or `UserInfoNotFoundError` I think?  would like to catch both.  I'm not 100% decided on if they need to be separate errors as you have vs a slightly more generic RedisKeyNotFoundError tbh.  Maybe both of these exceptions can inherit from that, and then this guy can catch the parent because he doesnt care about the distinction, but others might